contracts/Tipping_v2_Standalone.aes

```diff
@@ -110,25 +114,45 @@ contract Tipping =
     let url_id = get_url_id(url)
     let (last_claim_gen, amount, token_claims) = state.claims[url_id = (0, 0, {})]
     let tip_id = Map.size(state.tips)
-    let tip = { sender       = Call.caller
-              , title        = title
-              , claim_gen    = last_claim_gen + 1
-              , timestamp    = Chain.timestamp
-              , url_id       = url_id
-              , token        = Some(token)
-              , amount       = 0
-              , token_amount = token_amount }
+    let tip_meta = { sender    = Call.caller
+                   , title     = title
+                   , timestamp = Chain.timestamp }
+    let tip_token_data = { token  = token
+                         , amount = token_amount }
     let updated_token_claims = token_claims{ [token = 0] @ previous_amount = previous_amount + token_amount }
-    put(state{ tips[tip_id] = tip,
+    put(state{ tips[tip_id] = TokenTip(tip_meta, url_id, tip_token_data, last_claim_gen + 1),
                claims[url_id] = (last_claim_gen, amount, updated_token_claims) })
     token.transfer_allowance(Call.caller, Contract.address, token_amount)
     Chain.event(TipTokenReceived(Call.caller, token_amount, url, token))
     tip_id
 
+  payable stateful entrypoint tip_direct(receiver: receiver, title : string) : int =
```

Comment:
Should we change `title : string` to `title : option(string)` 🤔

Reply:
Eventually empty string is also fine for this, let's discuss this further